### PR TITLE
updates for full kit policies

### PR DESCRIPTION
**Pull request by cppvik**
* added rule to verify patcheck results

### CODE REVIEW

**Review by dn-scribe — APPROVED** [review]

File: policies/sarif/patcheck/updates-needed.yaml
Comment (dn-scribe) [inline policies/sarif/patcheck/updates-needed.yaml:1]: The tool name should be updated by Avi and and then here, but that could be delayed.

**Review by dn-scribe — APPROVED** [review]

**Review by dn-scribe — APPROVED** [review]

File: policies/slsa/verify-builder.rego [hunk 1]
```diff
@@ -34,11 +38,12 @@ reason = v {
...
 
-violations = j {
-	j := {r |
-		input.evidence.predicate.buildDefinition.internalParameters.hostname != input.config.args.id
-		r = {
-			"builder_id": input.evidence.predicate.buildDefinition.internalParameters.hostname,
-		}
-	}
+violations = [{"id": input.evidence.predicate.buildDefinition.internalParameters.run_id}] if {
+	input.evidence.predicate.buildDefinition.internalParameters.context_type == "github"
```
Comment (houdini91) [inline policies/slsa/verify-builder.rego:42]: why specifically for github.
I think == "local" should return violation (bulid should not be local), anything else, meaning 
github, circle, azure .. etc is considered a CI builder and you can compare the run_id (they all should have it populated)

File: policies/slsa/verify-builder.yaml [hunk 2]
```diff
@@ -9,8 +9,6 @@ initiatives:
   - client-policy
 
 evidence:
-  filter-by:
```
Comment (houdini91) [inline policies/slsa/verify-builder.yaml:12]: would you want this filter-by: pipeline ?

**Review by houdini91 — APPROVED** [review]

File: policies/slsa/verify-builder.yaml (hunk 2 above)
Comment (cppvik) [reply policies/slsa/verify-builder.yaml:12]: Good point, but in this case there's another issue, cause the rule error shows the same ID (from the SLSA provenance) that we have in the rule input.

Comment (houdini91) [reply policies/slsa/verify-builder.yaml:12]: maybe the ID should default to current.RULE_ID or something like that?

**Review by houdini91 — APPROVED** [review]

File: policies/slsa/verify-builder.yaml (hunk 2 above)
Comment (cppvik) [reply policies/slsa/verify-builder.yaml:12]: The meaning of this field is actually differs between a pipeline and a local environment. Maybe it's better to split them to different rules or at least to different configs, one defaulting to smth like `{{ github.run_id }}` and the other one leave templated.

File: policies/sarif/patcheck/updates-needed.yaml
Comment (cppvik) [reply policies/sarif/patcheck/updates-needed.yaml:1]: We can also set the toolname manually in the `evidence` command, but let's keep it like this for the demo.

**Pull request merged**